alchemy/src/neon/database.ts

```diff
@@ -0,0 +1,303 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+
+/**
+ * Properties for creating or updating a Neon database
+ */
+export interface NeonDatabaseProps extends NeonApiOptions {
+  /**
+   * The ID of the project containing the branch
+   */
+  project_id: string;
+
+  /**
+   * The ID of the branch where the database will be created
+   */
+  branch_id: string;
+
+  /**
+   * Name of the database
+   */
+  name: string;
+
+  /**
+   * Name of the role that will own the database
+   */
+  owner_name: string;
+
+  /**
+   * Whether to adopt an existing database if it already exists
+   * @default false
+   */
+  adopt?: boolean;
+}
+
+/**
+ * Response structure for database API operations
+ */
+export interface NeonDatabaseType {
+  /**
+   * The database ID
+   */
+  id: number;
+
+  /**
+   * The ID of the branch containing this database
+   */
+  branch_id: string;
+
+  /**
+   * Name of the database
+   */
+  name: string;
+
+  /**
+   * Name of the role that owns the database
+   */
+  owner_name: string;
+
+  /**
+   * Time at which the database was created
+   */
+  created_at: string;
+
+  /**
+   * Time at which the database was last updated
+   */
+  updated_at: string;
+}
+
+/**
+ * A Neon database within a branch
+ */
+export interface NeonDatabase
+  extends Resource<"neon::Database">,
+    Omit<NeonDatabaseProps, "apiKey"> {
+  /**
+   * The database ID
+   */
+  id: number;
+
+  /**
+   * The ID of the branch containing this database
+   */
+  branch_id: string;
+
+  /**
+   * Name of the database
+   */
+  name: string;
+
+  /**
+   * Name of the role that owns the database
+   */
+  owner_name: string;
+
+  /**
+   * Time at which the database was created
+   */
+  created_at: string;
+
+  /**
+   * Time at which the database was last updated
+   */
+  updated_at: string;
+}
+
+/**
+ * API response structure for database operations
+ */
+interface NeonDatabaseApiResponse {
+  database: NeonDatabaseType;
+}
+
+/**
+ * Payload structure for creating a database
+ */
+interface CreateDatabasePayload {
+  database: {
+    name: string;
+    owner_name: string;
+  };
+}
+
+/**
+ * Payload structure for updating a database
+ */
+interface UpdateDatabasePayload {
+  database: {
+    name: string;
+    owner_name: string;
+  };
+}
+
+/**
+ * API response structure for listing databases
+ */
+interface ListDatabasesResponse {
+  databases?: NeonDatabaseType[];
+}
```

Comment:
This is an internal type that is not needed outside of this file. Inline if possible. Move to bottom where the utility api functions are otherwise.